Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -0,0 +1,337 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Shared
+import UIKit
+import Deferred
+import Storage
+import WebImage
+
+
+// MARK: -  Lifecycle
+struct ASPanelUX {
+    static let backgroundColor = UIColor(white: 1.0, alpha: 0.5)
+    static let topSitesCacheSize = 20
+    static let historySize = 10
+}
+
+class ActivityStreamPanel: UIViewController, UICollectionViewDelegate {
+    weak var homePanelDelegate: HomePanelDelegate? = nil
+    let profile: Profile
+
+    lazy private var tableView: UITableView = {
+        let tableView = UITableView(frame: CGRect.zero, style: .Grouped)
+        tableView.registerClass(SimpleHighlightCell.self, forCellReuseIdentifier: "Cell")
+        tableView.registerClass(ASHorizontalScrollCell.self, forCellReuseIdentifier: "TopSite")
+        tableView.registerClass(HighlightCell.self, forCellReuseIdentifier: "Highlight")
+        tableView.backgroundColor = ASPanelUX.backgroundColor
+        tableView.separatorStyle = .None
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.rowHeight = UITableViewAutomaticDimension
+        tableView.separatorInset = UIEdgeInsetsZero
+        tableView.estimatedRowHeight = 65
+        tableView.estimatedSectionHeaderHeight = 15
+        tableView.sectionHeaderHeight = UITableViewAutomaticDimension
+        return tableView
+    }()
+
+    var topSiteHandler = ASHorizontalScrollSource()
+
+    //once things get fleshed out we can refactor and find a better home for these
+    var topSites: [TopSiteItem] = []
+    var history: [Site] = []
+
+
+    init(profile: Profile) {
+        self.profile = profile
+        super.init(nibName: nil, bundle: nil)
+
+        self.profile.history.setTopSitesCacheSize(Int32(ASPanelUX.topSitesCacheSize))
```

Comment:
Instead of the old topsites where the controller did view calculations to figure out the cache size. I just set a large enough default and then only display a certain amount. Keeps the VC cleaner.